extractuser.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"crypto/rsa"
+
+	"golang.org/x/net/context"
+
+	"net/http"
+
+	"github.com/almighty/almighty-core/token"
+	"github.com/goadesign/goa"
+)
+
+// ConfigureExtractUser is a configuration for middleware,
+// Accepts PublicKey and PrivateKey to create a token manager
+// Using token manager, it is responsible for extracting the token from context if present
+// Update the context with uuid found in token
+func ConfigureExtractUser(publicKey *rsa.PublicKey, privateKey *rsa.PrivateKey) (ExtractUser goa.Middleware) {
+	manager := token.NewManager(publicKey, privateKey)
+	return func(h goa.Handler) goa.Handler {
+		return func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+			// for now ignore the error becasue still test for logged in user is not done.
+			uuid, err := manager.Locate(ctx)
```

Comment:
@tsmaeder @aslakknutsen need advice, 
This middleware is unconditionally added in the service. So even if an API is not secured, this middleware will try to take out the token and resolve user.
One, should I find out a way to conditionally add middleware, something like Security("jwt") while designing.
Two, should I add something that will make it a "Guest User" (maybe empty UUID with all zeros in it).